DSC Resource/JustEnoughAdministration/JustEnoughAdministration.psm1

```diff
@@ -611,13 +620,152 @@ class JeaEndpoint
+            {
+                return $PSSessionConfiguration
+            }
+            else
+            {
+                return $null
+            }
+        }
+        else {
+            Write-Verbose 'WinRM service is not running. Cannot get PS Session Configuration(s).'
+            return $null
+        }
+    }
+
+    ## Unregister a PS Session Configuration based on its name
+    hidden [Void] UnregisterPSSessionConfiguration($Name)
+    {
+        $winRMService = Get-Service -Name 'WinRM'
+        if ($winRMService -and $winRMService.Status -eq 'Running')
+        {
+            # Temporary disabling Verbose as xxx-PSSessionConfiguration methods verbose messages are useless for DSC debugging
+            $VerbosePreferenceBackup = $Global:VerbosePreference
+            $Global:VerbosePreference = 'SilentlyContinue'
+            $null = Unregister-PSSessionConfiguration -Name $Name -Force -WarningAction 'SilentlyContinue'
+            $Global:VerbosePreference = $VerbosePreferenceBackup
+        }
+        else {
+            Throw "WinRM service is not running. Cannot unregister PS Session Configuration '$Name'."
+        }
+    }
+
+    ## Register a PS Session Configuration and handle a WinRM hanging situation
+    hidden [Void] RegisterPSSessionConfiguration($Name, $Path, $Timeout)
+    {
+        $winRMService = Get-Service -Name 'WinRM'
+        if ($winRMService -and $winRMService.Status -eq 'Running')
+        {
+            Write-Verbose "Will register PSSessionConfiguration with argument: Name = '$Name', Path = '$Path' and Timeout = '$Timeout'"
+            # Register-PSSessionConfiguration has been hanging because the WinRM service is stuck in Stopping state
+            # therefore we need to run Register-PSSessionConfiguration within a job to allow us to handle a hanging WinRM service
+
+            # Save the list of services sharing the same process as WinRM in case we have to restart them
+            $processId = Get-CimInstance -ClassName 'Win32_Service' -Filter "Name LIKE 'WinRM'" | Select-Object -Expand 'ProcessId'
+            $serviceList = @(Get-CimInstance -ClassName 'Win32_Service' -Filter "ProcessId=$processId" | Select-Object -Expand 'Name')
+            foreach ($service in $serviceList.clone())
+            {
+                $dependentServiceList = Get-Service -Name $service | ForEach-Object { $_.DependentServices }
+                foreach ($dependentService in $dependentServiceList)
+                {
+                    if ($dependentService.Status -eq 'Running' -and $serviceList -notcontains $dependentService.Name)
+                    {
+                        $serviceList += $dependentService.Name
+                    }
+                }
+            }
+
+            if ($Path)
+            {
+                $registerString = "`$null = Register-PSSessionConfiguration -Name '$Name' -Path '$Path' -Force -ErrorAction 'Stop' -WarningAction 'SilentlyContinue'"
```

Comment:
Can we use a static script block with paramenters and the argument list to Start-job?